EIPS/eip-network_upgrade_windows.md

```diff
@@ -0,0 +1,171 @@
+Possible examples may include roadmap upgrades that need to occur in multiple
+upgrades or for security risks that have a existing mitigation in place that
+would be better served by a network upgrade. Another possible reason may be to
+defuse the difficulty bomb due to postponed roadmap upgrades.
+
+Priority network upgrades are best launched in unused roadmap launch windows,
+namely the third week of January, April, July, and October. If necessary they
+may be launched in the third week of any month, but strong consideration and
+preference should be given to unused roadmap launch windows.
+
+Priority network upgrades should be announced and a block chosen far enough in
+advance so major clients implementors can release software with the needed block
+number in a timely fashion. It is expected that two to four weeks would be
+sufficient.
+
+### Critical Network Upgrades
+
+Critical network upgrades are network upgrades that are designed to address
+systemic risks to the protocol or to the ecosystem. Historical examples include
+Dao Fork, Tangerine Whistle, and Spurious Dragon.
+
+This EIP provides neither guidance nor restrictions to the development and
+deployment of these emergency hard forks. These upgrades are typically launched
+promptly after a solution to the systemic risk is agreed upon between the client
+implementors.
+
+### Network Upgrade Naming
+
+The choice of a name for a network upgrade can help communicate what type of
+network upgrade it is. There have been patterns used in the past that if
+continued would help communicate this.
+
+Roadmap Network Upgrades have historically been given single word names. The
+first three network upgrades were Frontier, Homestead, and Metropolis.
+Metropolis was split into two upgrades, Byzantium and Constantinople. As of the
+writing of this EIP the next network upgrade will be called Istanbul.
+
+Critical Network Upgrades have historically been two word phrases. Sometimes the
+name applies to the issue at hand, sometimes it do not. Previous Critical
+Network upgrade names have been Dao Fork, Tangerine Whistle, Spurious Monkey,
+and Constantinople Fix.
+
+### Network Upgrade Block Number Choice
+
+When choosing an activation block the number can be used to communicate the role
+of a particular network in the Ethereum Ecosystem.
+
+To date all Mainnet activation blocks have ended in three or more zeros,
+including Critical Network Upgrades. Ropsten and Kovan initially started with
+three zeros but switched to palindromic numbers. Rinkeby has always had
+palindromic activation blocks. Goerli has yet to perform a network upgrade.
+
+To continue this pattern network upgrade activation block numbers for mainnet
+deployment should chose a number whose base 10 representation ends with three or
+more zeros.
+
+For testnet deployments network operators are encouraged to choose a block
+activation number that is a palindrome in base 10.
```

Comment:
Kind of random, but I could see why it might be useful to make this distinction. Warrants a little further discussion why if this section is kept.

Reply:
Choosing a block number is part of getting into the launch window.  So it felt in-scope.

Reply:
Yeah, but how people choose block numbers is up to them. It only matters to this proposal that they choose _a_ number and align everyone on that date.